### ISSUE TITLE
Make sure that all focusable items are instrumented with a phetioID

### ISSUE THREAD

**Issue by zepumph** [phetsims/faradays-law#83]
in https://github.com/phetsims/joist/issues/445#issuecomment-340614487 I implemented a version of the FocusProperty that has indices (like a trail to the node) and a phetioID, but some tab navigable items aren't instrumented. In Faradays Law, I think it is just the about dialog and the magnet. @samreid how should we go about instrumenting the focushighlight node around the magnet? I'm not sure how to hard code it as the same name as the tandem magnet itself, since it is just for display.

**Comment by samreid** [comment]
> @samreid how should we go about instrumenting the focushighlight node around the magnet?

It seems the MagnetNodeWithField should have a phetioID and receive the focus.  Why would the highlight node need to be instrumented?

**Comment by zepumph** [comment]
Sorry not the highlight node, but the node that the focusHighlight exists on, in this case it is quite the "implementation detail" as it is on the MagnetNodeWithField's `draggableNode`:

```js
 // the draggable container for the magnet and arrows
    var draggableNode = new Node( { cursor: 'pointer' } );
    this.addChild( draggableNode );

    // magnet
    self.magnetNode = createMagnetNode( model.magnetModel );
    draggableNode.addChild( self.magnetNode );

    // a11y
    draggableNode.tagName = 'div';
    draggableNode.ariaRole = 'application';
    draggableNode.focusable = true;
    draggableNode.focusHighlightLayerable = true;
    var draggableNodeFocusHighlight = new FocusHighlightPath( new Shape() ); // overridden once the draggableNode is fully constructed
    draggableNode.focusHighlight = draggableNodeFocusHighlight;
    
    // I added this line to accomplish this issue
    draggableNode.mutate( { tandem: tandem.createTandem( 'draggableNode' ) } );

```

By adding the bottom line, I was able to fill in the `focusedPhetioID` key filled in for the magnet:
![image](https://user-images.githubusercontent.com/6856943/32340608-eb32a7a0-bfaf-11e7-8cce-c601bc915034.png)

**Comment by samreid** [comment]
I can't think of a way to get away from the implementation detail, not sure what's best.  Best I can think of at the moment is to give it a name similar to but different from the magnet.  But that sounds like leaking implementation details.

**Comment by samreid** [comment]
I think there is an a11y way around this. Nodes can define their own focus region and input handlers, perhaps we should investigate making the magnet with field node be the focused thing and not this surrogate?

**Comment by zepumph** [comment]
Now that the focusProperty is out of instance proxies when a11y is not enabled, perhaps we should come back to this when a11y work continues in this repo. I say this because since 19 days ago #89 has completely reimplemented the keyboard navigation, and this issue could look very different depending on how the design process goes.

**Comment by zepumph** [comment]
This seems like a subset of https://github.com/phetsims/phet-io/issues/1205, unassigning to be worked on there.

**Comment by samreid** [comment]
@zepumph can we add an assertion, something like:

```js
if (window.phet.phetio && accessibilityEnabled && this.isFocusable){
  assert && assert(this.tandem.isLegalAndUsable(),'When running phet-io mode, all focusable instances must be instrumented.');
}
```

**Comment by zepumph** [comment]
@jessegreenberg, @mbarlow12 and I have been discussing assertions like that a lot lately. We are not exited about breaking a11y in all sims that have not had a11y work done to them, maybe this is not reasonable. It is a different approach to from phet-io, which currently is broken in all sims, and must have a query parameter added (`phetioValidateTandems=false`) for the sim to run. 

We could probably implement the above assertion orthogonally of the larger conversation though because it is only for phet-io brand.

**Comment by zepumph** [comment]
Proposed solution is in `Display.js`, not committed:
```js
    set focus( value ) {

      // If in phet-io brand, a11y is enabled, and the focus is not null
      if( window.phet && phet.phetio && phet.chipper.a11yEnabled && value ){
        var node = value.trail.lastNode();
        assert && assert( node.tandem && node.tandem.isLegalAndUsable(), 
          'When running phet-io mode, all focusable instances must be instrumented.');
      }
      // . . .
```

@samreid I'm not sure how to access the tandem consistently from a Node, is there a way like above?

**Comment by samreid** [comment]
Yes, `Node.tandem` is defined on line 3611

**Comment by zepumph** [comment]
This seems like a good way to go, but what about things we have been specifically deinstrumenting, like the about dialog. If we instrument it again, would it just be an ObjectIO? That way no NodeIO methods could be called on it, or should it just be phetioReadOnly anyways?

**Comment by zepumph** [comment]
Each link will probably need to be instrumented then also. 
![image](https://user-images.githubusercontent.com/6856943/33285089-bbca4bc8-d35d-11e7-9475-4745454e8d35.png)

**Comment by zepumph** [comment]
I did a few things here. 
1. Instrumented the about dialog as phetioReadOnly since it has focusable entities in it.
2. Instrumented the `draggableNode` as a child of the magnet ( I don't love this)
3. Added the check in Display's `set focus` to assert if phet-io + a11y and you try to focus to an uninstrumented item. 

Many of the changes above have strange phetioReadOnly option passing. A new issue was created to discuss that.
@samreid please review.

**Comment by samreid** [comment]
The preceding changes seem reasonable and I cannot think of another way to deal with showing focus during playback of input events.  Closing.
